Server/src/models/DataUploadModel.ts

```diff
@@ -15,190 +15,185 @@ import { Material } from './entities/Material';
     }
-}
 
-const insertPublication = async (referenceTitle, referencePages, preferenceTypeID, publisherNameId, referenceYear, referenceVolume, referenceAuthors) => {
-
-    const connection = getConnection();
-
-    let publication = new Publications();
-    publication.id;
-    publication.name = referenceTitle;
-    publication.doi;
-    publication.pages = referencePages;
-    publication.publicationtypeId = preferenceTypeID;
-    publication.publisherId = publisherNameId;
-    publication.year = referenceYear;
-    publication.volume = referenceVolume;
-    publication.datePublished;
-    publication.dateAccessed;
-    publication.authors = referenceAuthors;
-    await connection.manager.save(publication);
-    console.log("publication saved.");
-    return publication.id;
-}
+    async insertPublication(referenceTitle: string, referencePages: number, preferenceTypeID: number, publisherNameId: number, referenceYear: number, referenceVolume: number, referenceAuthors: any[]) {
+
+        const connection = getConnection();
+
+        let publication = new Publications();
+        publication.id;
+        publication.name = referenceTitle;
+        publication.doi;
+        publication.pages = referencePages;
+        publication.publicationtypeId = preferenceTypeID;
+        publication.publisherId = publisherNameId;
+        publication.year = referenceYear;
+        publication.volume = referenceVolume;
+        publication.datePublished;
+        publication.dateAccessed;
+        publication.authors = referenceAuthors;
+        await connection.manager.save(publication);
+        console.log("publication saved.");
+        return publication.id;
+    }
 
-const insertMaterial = async (material) => {
+    async insertMaterial(material: any[]) {
 
-    const connection = getConnection();
+        const connection = getConnection();
 
-    for (var i = 0; i < material.length; i++) {
-        let composition = new Composition();
-        composition.id;
-        composition.composition = material[i].composition;
-        await connection.manager.save(composition);
+        for (var i = 0; i < material.length; i++) {
+            let composition = new Composition();
+            composition.id;
+            composition.composition = material[i].composition;
+            await connection.manager.save(composition);
 
-        let someMaterial = new Material();
-        someMaterial.id;
-        someMaterial.compositionId = composition.id;
-        someMaterial.details = material[i].details;
-        await connection.manager.save(someMaterial);
+            let someMaterial = new Material();
+            someMaterial.id;
+            someMaterial.compositionId = composition.id;
+            someMaterial.details = material[i].details;
+            await connection.manager.save(someMaterial);
+        }
     }
-}
 
-const insertCategories = async (_category, sub_category) => {
+    async insertCategories(_category: string, _subCategory: string) {
```

Comment:
whats up with underscores

Reply:
Either variable or param got the underscore, I gave it to the param to keep category there

Reply:
I didnt want to change category so I put it in the param

Reply:
I dont see why there needs to be a separation at all tbh, why dont all of them start with letters, instead of underscores, does it improve readability in your opinion?